### PR TITLE
CAD-1347 shelley3pools:  switch to genesis-specified pools

### PR DESCRIPTION
**Pull request by deepfire**
1. Bump the node for the new CLI dependencies & serialisation formats.
1. Update generator code to the new dependencies.
1. Provide genesis-specified declarative staking definition genesis generation (old genesis generation still available).

### CODE REVIEW

File: tx-generator-shelley/src/Cardano/Benchmarking/TxGenerator/Phase1.hs [hunk 1]
```diff
@@ -4,42 +4,64 @@ where
...
-    splitVal = Lovelace $ fromIntegral (value - (T.unFeePerTx $ P.fee args) ) `div` fromIntegral txCount
+    splitVal = Lovelace $ fromIntegral (value - (T.unFeePerTx $ P.fee args) )
+                                           `div` fromIntegral txCount
     splitList :: [Lovelace]
     splitList = replicate txCount splitVal
-    network = P.network args
-    mapSubmitError e = case e of
+          
+    castTxIn :: UApi.TxIn -> TxIn
+    castTxIn = error "castTxIn"
```
Comment (MarcFontaine) [inline tx-generator-shelley/src/Cardano/Benchmarking/TxGenerator/Phase1.hs:50]: TODO: Implement this

File: tx-generator-shelley/src/Cardano/Benchmarking/TxGenerator/Phase1.hs [hunk 2]
```diff
@@ -4,42 +4,64 @@ where
...
          , Producer.fund = Lovelace $ fromIntegral value
          }
   (tx,p) <- withExceptT Phase1SplitError $ hoistEither $ Producer.split splitList initialFund
-  newExceptT $ fmap mapSubmitError $ Api.submitTx network (P.socketPath args) tx
+  newExceptT $ fmap mapSubmitError $ Api.submitTx connectInfo $ TxForShelleyMode tx
   return p
   where
-    (P.InitialFund value keyFile utxo address) = P.initialFund args
+    thisNetwork = P.network args
+    connectInfo = error "connectInfo"
```
Comment (MarcFontaine) [inline tx-generator-shelley/src/Cardano/Benchmarking/TxGenerator/Phase1.hs:41]: TODO:implement this

**Review by MarcFontaine — APPROVED** [review]
The Haskell changes look good to me.

**Review by denisshevchenko — APPROVED** [review]

**Pull request merged**